music/src/core/sequences.ts

```diff
@@ -424,3 +424,95 @@ export function mergeInstruments(ns: INoteSequence) {
 
   return result;
 }
+
+/**
+ * Splits an unquantized `NoteSequence` into chunks.
+ *
+ * @param ns The `NoteSequence` to split.
+ * @param chunkSize The number of steps to split the note sequence in.
+ * For example, if you want to split the sequence into 2 bar chunks, then
+ * if the sequence has 4 steps/quarter, that will be 32 steps for each 2 bars.
+ *
+ * @returns An array of `NoteSequences` each of which are at most `chunkSize`
+ * steps.
+ */
+export function split(seq: INoteSequence, chunkSize = 32): NoteSequence[] {
+  assertIsQuantizedSequence(seq);
+
+  // Make a clone so that we don't destroy the input.
+  const ns = clone(seq);
+
+  // Sort notes first.
+  const notesBystartStep =
+      ns.notes.sort((a, b) => a.quantizedStartStep - b.quantizedStartStep);
+
+  const chunks = [];
+  let startStep = 0;
+  let currentNotes = [];
+
+  for (let i = 0; i < notesBystartStep.length; i++) {
+    const note = notesBystartStep[i];
+
+    const originalStartStep = note.quantizedStartStep;
+    const originalEndStep = note.quantizedEndStep;
+
+    // Rebase this note on the current chunk.
+    note.quantizedStartStep -= startStep;
+    note.quantizedEndStep -= startStep;
+
+    if (note.quantizedStartStep < 0) {
+      continue;
+    }
+    // If this note fits in the chunk, add it to the current sequence.
+    if (note.quantizedEndStep <= chunkSize) {
+      currentNotes.push(note);
+    } else {
+      // If this note spills over, truncate it and add it to this sequence.
+      if (note.quantizedStartStep < chunkSize) {
+        currentNotes.push(new NoteSequence.Note({
+          pitch: note.pitch,
```

Comment:
I worry that you may only copy a subset of the fields, for example if we add another later. Can you just copy the whole note and then change the steps?